### PR TITLE
[Core] Fix generate edges in geometries

### PR DESCRIPTION
**Pull request by ddiezrod**
I found that in most geometries edges are not created with the correct numbering.

For example in the line with 3 points, the order should be: 0 ------ 2 ------ 1, but in most cases  they are being created with 0----1----2, which leads to several problems (for example, length is not correctly computed)

### CODE REVIEW

File: kratos/geometries/triangle_2d_10.h [hunk 1]
```diff
@@ -693,9 +693,9 @@ class Triangle2D10 : public Geometry<TPointType>
     GeometriesArrayType GenerateEdges() const override
     {
         GeometriesArrayType edges = GeometriesArrayType();
-        edges.push_back(Kratos::make_shared<EdgeType>(this->pGetPoint(0), this->pGetPoint(3), this->pGetPoint(4), this->pGetPoint(1)));
```
Comment (loumalouomega) [inline kratos/geometries/triangle_2d_10.h:696]: These implementations are new @mnabideltares, is it correct?

**Review by loumalouomega — DISMISSED** [review]
Wait for @mnabideltares, a priori the new elements should be right

File: kratos/geometries/triangle_2d_10.h (hunk 1 above)
Comment (ddiezrod) [reply kratos/geometries/triangle_2d_10.h:696]: They had the same mistake as the other ones I think.

**Comment by ddiezrod** [comment]
I did not know there was already an issue for this. 

https://github.com/KratosMultiphysics/Kratos/issues/9422

I understand that what I am doing here goes in the same line of what was discussed there. @Vahid-Galavi  thanks for pointing me to the issue.

**Comment by Vahid-Galavi** [comment]
You are welcome @ddiezrod

File: kratos/geometries/triangle_2d_10.h (hunk 1 above)
Comment (mnabideltares) [reply kratos/geometries/triangle_2d_10.h:696]: I followed the same reciepe of other high order elements. So, it needs to be investigted too. However, during implementation of these elements, my test cases all worked as expected with this definition of points

Comment (mnabideltares) [reply kratos/geometries/triangle_2d_10.h:696]: However, my test cases all worked correctly with this definition of points

Comment (mnabideltares) [reply kratos/geometries/triangle_2d_10.h:696]: @loumalouomega  and @ddiezrod 
Do I modify the the nodes order for 10N and 15N elements?

Comment (ddiezrod) [reply kratos/geometries/triangle_2d_10.h:696]: I think the order is correct now with the changes in this branch, isn't it?

Comment (mnabideltares) [reply kratos/geometries/triangle_2d_10.h:696]: Yes, it seems to be correct. Thank you

**Comment by ddiezrod** [comment]
Tests are passing now with the change in @KratosMultiphysics/geomechanics , is this ok for everyone?

**Review by loumalouomega — APPROVED** [review]

**Comment by loumalouomega** [comment]
From my side yes

**Comment by mcgicjn2** [comment]
Yes these changes are fine from our side in GeoMechanicsApplication - Thanks for doing it

**Pull request merged**